src/main/scala/com/twitter/finagle/postgres/PostgresClient.scala

```diff
@@ -74,6 +75,21 @@ trait PostgresClient {
    * with a reasonable likelihood of success).
    */
   def isAvailable: Boolean
+
+  /**
+    * Starts listening to the given channel and registers the consumer
+    * @param channel the name of the channel
+    * @param block the consumer that accepts a [[NotificationResponse]]
+    * @return a function to stop consuming notifications
+    */
+  def listen(channel: String, block: NotificationResponse => Unit): Future[Runnable]
```

Comment:
Can you describe the abstraction here a bit? Why does this return `Future[Runnable]`?

Reply:
The idea is to return a function that unregisters the consumer for/if the client wants to stop receiving events. Another approach can be to add another method to the client to do that. What do you think about it?

Reply:
I am not quite sure what the abstraction should be. But `Future[Runnable]` seems off base to me. It probably deserves its own variety of handle type, like `Listener` or something.

It would also be cool if this would return `AsyncStream` of notification payloads (seems more finagle-y). But `AsyncStream` doesn't have a built-in `close` interface, so we'd need to make up for that somehow.

Reply:
I first thought about a stream. But it seemed more straightforward and easier for a client to use callback semantics. I started working on the `AsyncStream` version. I'm a little busy this days, but I'll try to get back ASAP. Thanks for the feedback!

Reply:
Well, it doesn't necessarily have to be `AsyncStream`. A callback-based mechanism might be OK too. Maybe we can start with that and then implement an `AsyncStream` option later, in addition.

For a callback-based system, maybe `listen` could accept a `Service` to call with the `NotificationResponse`? And instead of returning `Future[Runnable]`, maybe it could just return `Unit` (user must call `unlisten` - better symmetry with Postgres)? Another option might be for it to return its own `Future[Listener]`, where `Listener` has an `unlisten` method - but I think if you return a different "Listener" for each call, it's a bit disingenuous because multiple `listen`s for the same channel will all be unlistened with a single `unlisten` call. That's why I think `Unit` might be a better return type for this.